tests/frame/unpivot_test.py

```diff
@@ -70,3 +76,26 @@ def test_unpivot_default_var_value_names(constructor: Constructor) -> None:
     result = df.unpivot(on=["b", "c"], index=["a"])
 
     assert result.collect_schema().names()[-2:] == ["variable", "value"]
+
+
+@pytest.mark.parametrize(
+    ("data", "expected_dtypes"),
+    [
+        (
+            {"idx": [0, 1], "a": [1, 2], "b": [1.5, 2.5]},
+            [nw.Int64(), nw.String(), nw.Float64()],
+        ),
+    ],
+)
+def test_unpivot_mixed_types(
+    request: pytest.FixtureRequest,
+    constructor: Constructor,
+    data: dict[str, Any],
+    expected_dtypes: list[DType],
+) -> None:
+    if "dask" in str(constructor):
+        request.applymarker(pytest.mark.xfail)
```

Comment:
Dask schema is: `[nw.Int64, nw.Object, nw.Float64]`. If we want to be more flexible in tests like this, we could adopt [dirty-equals](https://github.com/samuelcolvin/dirty-equals) - I really like the tool, I both used it and contributed to it in the past.